src/db/migrations/20240108008813_marketing_attributions.js

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+export async function up(knex) {
+  return knex.schema
+    .createTable("attributions", table => {
+      table.increments('id').primary()
+      table.integer("subscriber_id").references("subscribers.id").notNullable().onDelete("CASCADE").onUpdate("CASCADE");
+      table.string("type").notNullable()
+      table.string("utm_source")
+      table.string("utm_campaign")
+      table.string("utm_medium")
+      table.string("utm_term")
+      table.string("entrypoint")
+      table.string("flow_id")
```

Comment:
Is storing the `flow_id` useful? I thought this was for an individual session, it seems like we could potentially have a lot more rows if we store this.

Reply:
@rhelmer One thing I need to add is clearing the cookies once they are preserved in the db
I'm not sure how permanent these rows should be... I think we can keep them forever, or we can actively delete them once glean events are submitted?

Reply:
Another option I was thinking about is adding a UNIQUE on `subscriber_id + type` so there's a limit of max 2 entries per `subscriber_id`.. instead of appending, we will just update. That's also an option to keep the record count low

Reply:
I think it's probably worth recording any UTM parameters encountered, it's likely that the attribution model will get more sophisticated in the future, so it might be nice to have the data there. I'm not too worried about the row count but `flow_id` I just don't know if it's worth storing since it's supposed to be a per-session thing.

Reply:
> @rhelmer One thing I need to add is clearing the cookies once they are preserved in the db I'm not sure how permanent these rows should be... I think we can keep them forever, or we can actively delete them once glean events are submitted?

I think we should keep the data in the database, I'm not worried about the cookies after we've used them,

Reply:
makes sense, I can take that out

Reply:
> > @rhelmer One thing I need to add is clearing the cookies once they are preserved in the db I'm not sure how permanent these rows should be... I think we can keep them forever, or we can actively delete them once glean events are submitted?
> 
> I think we should keep the data in the database, I'm not worried about the cookies after we've used them,

Another related question is, do we want to set a unique on the table for the utm params:
Say we have a match of 
```
"subscriber_id"
      "type"
     "utm_source"
      "utm_campaign"
      "utm_medium"
      "utm_term"
      "entrypoint"
      ```
      
      Would we ever want to record that again in the db when a record already exists in the past?

Reply:
No I don't think we need to store duplicates... in fact the more I think about it, we're already emitting telemetry events with this data so we probably don't have to worry too much about permanent storage in any case.

Reply:
I'm setting an unique() on the table: https://github.com/mozilla/blurts-server/pull/3994/commits/543f182963afa11b194ef115ef92ecb87cda3623